Module api RM_CallArgv and test

oranagra opened the pull request:
This API is mainly useful in case you wanna forward a module command to another
one using RM_Call.

Adding a test for coverage for RM_Call and RM_CallArgv in a new "misc" unit
to be used for various short simple tests

antirez commented:
Hi @oranagra, please could you rebase? Thanks.

oranagra commented:
@antirez rebased, and also solved some compilation warnings (const in redismodule.h and missing usleep in fork.c)

antirez commented:
@oranagra thank you. Given that in this new API, you still get the command name as a separated argument, what is the main difference with just using the `"v"` format of RM_Call()?

oranagra commented:
@antirez you're perfectly right. i was not aware of it, i'll drop that API.
considering this PR has some test coverage and warning fixes, instead of completely rejecting it, i just reverted all the code about RM_CallARV and refactoring, adjusted the test code to use the 'v' format, and edited the commit comment.

so this now has nothing to do with the original PR title, but please consider merging other changes.

antirez commented:
@oranagra is it possible that we are missing something? Because during a meeting in Tel Aviv I said the same thing, that we had "v", and the original proposer of the feature replied with some argument that made sense IIRC.

oranagra commented:
@antirez i'll shake the tree and see if anyone remembers. @guybe7 @yossigo 
i did implement the same test using the 'v' and it works, so maybe the original argument was not about forwarding a command at all...

yossigo commented:
@oranagra I looked up the history and didn't come up with anything conclusive, but I think this API was created before the `'v'` option was implemented.

oranagra commented:
@antirez let's forget about RM_CallArgv, but please merge this PR that improves coverage and solves warnings (trimmed the new API from both code and commit comment)

antirez commented:
@yossigo the "v" option exists since the very start AFAIK. I think that simply the fact that RM_Call() is different in the way *it is documented* makes the difference. It hasn't a top-comment and is not part of the reference. It's documented in the modules official getting started doc.

antirez commented:
@oranagra Thanks Oran, PR merged.
